### PR TITLE
[MERGE 1st DEC] Changes to UK visa questions for EU citizens

### PR DESCRIPTION
**Pull request by ChrisBAshton**
Adds a new question for EU citizens: 'When are you coming to the
UK?'. If they answer 'before 1 January 2021', take them to an
outcome that says they do not need a visa. We could have used the
existing outcome_no_visa_needed.erb template here but we may want
to write something bespoke and having a separate template allows
us to do that more easily.

If citizens answer 'from 1 January 2021', route them to the
existing flow for non-visa nationals, for example Argentina:
https://www.gov.uk/check-uk-visa/y/argentina

Trello: https://trello.com/c/WoFhTMCi/2196-5-1-dec-logic-changes-for-eu-citizens-check-if-you-need-a-uk-visa

---

:warning: This application is Continuously Deployed: :warning:

- Merged changes are automatically deployed to staging and production.

- Make sure you follow [the guidance for deployments](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) **before** you merge.

- Check your branch is being deployed in the [Release app](https://release.publishing.service.gov.uk/applications/smartanswers), after merging.

- Changes to start pages are **not** continuously deployed, and are [deployed using a different process](https://github.com/alphagov/smart-answers/blob/master/doc/smart-answer-flow-development/publishing.md).

### CODE REVIEW

File: lib/smart_answer_flows/check-uk-visa.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def define
           elsif calculator.passport_country_is_macao?
             question :what_sort_of_travel_document?
           elsif calculator.passport_country_in_eea?
-            outcome :outcome_no_visa_needed
+            outcome :when_are_you_coming_to_the_uk?
```
Comment (theseanything) [inline lib/smart_answer_flows/check-uk-visa.rb:34]:
```suggestion
            question :when_are_you_coming_to_the_uk?
```

**Comment by theseanything** [comment]
Hey @ChrisBAshton - just wondering does this PR supersede https://github.com/alphagov/smart-answers/pull/4174 ?

**Review by thomasleese — APPROVED** [review]

**Comment by ChrisBAshton** [comment]
(Pushed up to rebase against master, so we have a cleaner diff in #4995. Nothing has changed since the code was approved)

**Comment by ChrisBAshton** [comment]
Merged #4995 ready to deploy this PR. There are, however, a couple of final tweaks to be made before we can deploy tomorrow.

Any changes should appear as separate commits below this comment. All of the commits above this line have already been approved so should not need a re-review.

**Review by MuriloDalRi — APPROVED** [review]

**Review by callumknights — APPROVED** [review]

**Pull request merged**